connect.go

```diff
@@ -364,7 +364,7 @@ func receiveUnaryResponse[T any](conn StreamingClientConn) (*Response[T], error)
 	// In a well-formed stream, the response message may be followed by a block
 	// of in-stream trailers or HTTP trailers. To ensure that we receive the
 	// trailers, try to read another message from the stream.
-	if err := conn.Receive(new(T)); err == nil {
```

Comment:
I think we can remove this extra Receive call with this protocol drain fix: https://github.com/bufbuild/connect-go/pull/536

Reply:
Awesome. That'd also let us make the function non-generic, which would cut binary size a bit.